### PR TITLE
WIP rerender error

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: .ci_support/migrations/cuda111_112.yaml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      - quay.io/condaforge/linux-anvil-cuda:10.0    # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
+      - quay.io/condaforge/linux-anvil-cuda:10.1    # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
+      - quay.io/condaforge/linux-anvil-cuda:10.2    # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
+      - quay.io/condaforge/linux-anvil-cuda:11.0    # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
+      - quay.io/condaforge/linux-anvil-cuda:11.1    # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
+      - quay.io/condaforge/linux-anvil-cuda:11.2    # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
+      - quay.io/condaforge/linux-anvil-cos7-cuda:9.2     # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
+      - quay.io/condaforge/linux-anvil-cos7-cuda:10.0    # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
+      - quay.io/condaforge/linux-anvil-cos7-cuda:10.1    # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
+      - quay.io/condaforge/linux-anvil-cos7-cuda:10.2    # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
```
Comment (beckermr) [inline .ci_support/migrations/cuda111_112.yaml:35]: I added only the extra ordering keys here.

File: recipe/conda_build_config.yaml [hunk 2]
```diff
@@ -19,3 +19,23 @@ cuda_compiler_version:
...
+  - cos6   # [linux64]
+  - cos7   # [linux and aarch64]
+  - cos7   # [linux and ppc64le]
+  - cos7   # [linux and armv7l]
+
+  - cos7   # [linux64]
+  - cos7   # [linux64]
+  - cos7   # [linux64]
+  - cos7   # [linux64]
+  - cos6   # [linux64]
```
Comment (beckermr) [inline recipe/conda_build_config.yaml:41]: I put the extra zipping keys here.

**Comment by beckermr** [comment]
ok @leofang @isuruf I have tried the changes discussed on gitter here. I am stumped. 

I added only the ordering keys and fixed the zipping. The feedstock rerenders, but the result doesn't add 11.1 or 11.2

File: .ci_support/linux_64_cuda_compiler_version10.2.yaml [hunk 3]
```diff
@@ -13,9 +13,10 @@ cuda_compiler:
 cuda_compiler_version:
 - '10.2'
 docker_image:
-- quay.io/condaforge/linux-anvil-cos7-cuda:10.2
+- quay.io/condaforge/linux-anvil-cos7-cuda:9.2
```
Comment (beckermr) [inline .ci_support/linux_64_cuda_compiler_version10.2.yaml:16]: Also this is wrong.

**Comment by beckermr** [comment]
One tidbit here is that smithy is ignoring the ordering key in the migrator. :/

**Comment by beckermr** [comment]
That key is only used for special key_add or key_remove operations in the migrators. cuda111_112.yaml is not one of those.

**Comment by beckermr** [comment]
I think there is a bug in smithy where the CBC from the feedstock is applied twice. 

I also generalize key adds to work for more than one set of zip keys.

**Comment by leofang** [comment]
Just FYI, starting cuDNN 8.1.0 for all CUDA 11.x there is a single source release. I haven't examined the content in the new package, but I suspect similar to cuTENSOR it also utilizes the binary compatibility feature on 11.x, so perhaps it's better to follow what I've done in cutensor-feedstock and release 1 conda package for all CUDA 11.x, saving us some headache here. The price to pay is to maintain a conda-forge-config.yml and reject migrators.

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
New smithy is working fine here. I am going to close the PR but leave my branch. If people want these changes, we can put them into the pinnings repo and a new PR on cudnn

**Pull request closed without merging**